### ISSUE TITLE
mod_admin_extra private_set example not working

### ISSUE THREAD

**Issue by mrfrostfire** [processone/ejabberd#1057]
We are using ejabberd 16.02 on red hat linux.

When trying to use the following private_set example from mod_admin_extra.erl file:
%% Example usage:
%% $ ejabberdctl private_set badlop localhost "\<aa\ xmlns=\'bb\'\>Cluth\</aa\>"
%% $ ejabberdctl private_get badlop localhost aa bb
%% <aa xmlns='bb'>Cluth</aa>

The private_set command gives the error:
Error: the command "private_set" requires 1 less argument.

**Comment by mrfrostfire** [comment]
I can add that i've also tried setting the value as a variable (with and without quotes in the variable) then send the variable to the private_set command, and  it still gives the same error

**Comment by badlop** [comment]
There were changes related to ejabberdctl escaping in the latest months, maybe one of them solved this. Right now, using git master, the comand works correctly this way:

```
$ ejabberdctl private_set badlop localhost "<aa\ xmlns=\'bb\'>Cluth</aa>"
Error found parsing the element:
  <<"<aa\\ xmlns=\\'bb\\'>Cluth</aa>">>
Error: {4,<<"not well-formed (invalid token)">>}

$ ejabberdctl private_set badlop localhost "<aa xmlns='bb'>Cluth</aa>"

$ ejabberdctl private_get badlop localhost aa bb
<aa xmlns='bb'>Cluth</aa>
```

**Comment by mrfrostfire** [comment]
great, then in the next release it might be solved.
I tried $ ejabberdctl private_set badlop localhost "<aa xmlns='bb'>Cluth</aa>" on our current 16.02 and it gives me the same 1 less argument error.
Obviously I tried with real username and host as well

**Comment by badlop** [comment]
The commands work correctly for me using ejabberd git 16.02, so maybe the problem is not solved for you with a new release.

In the worrst case, you can use other method, like enabling ejabberd_xmlrpc and using a Python script to call the command.

**Comment by mrfrostfire** [comment]
We ran into another obstacle causing us to use pubsub instead,
Now we have another issue with pubsub and hometree instead. We have traced
the issue to where it breaks and we think it is an issue in the ejabberd
code.
That is offtopic for this ticket however.
Is there any way to continue that discussion or is the only way to open a
new ticket?
Thanks
Patrick
mån 11 apr. 2016 kl. 13:22 skrev badlop <notifications@github.com>:
